src/modules/flow/iio/nodes.c

```diff
@@ -184,9 +182,9 @@ gyroscope_close(struct sol_flow_node *node, void *data)
 {
     struct iio_direction_vector_data *mdata = data;
 
-    free((char *)mdata->config.trigger_name);
-    if (mdata->device)
-        sol_iio_close(mdata->device);
+    free((char *)mdata->iio_base.config.trigger_name);
+    if (mdata->iio_base.device)
+        sol_iio_close(mdata->iio_base.device);
```

Comment:
Thanks @edersondisouza, just cast the void * to the base struct. I will change the close/tick functions as you said